<a href="https://colab.research.google.com/github/rishh19/FlyRank-AI-Internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
import os
import sys
import subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/rishh19/FlyRank-AI-Internship"
REPO_DIR = "FlyRank-AI-Internship"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)

print("Current directory:", os.getcwd())

Current directory: /content/FlyRank-AI-Internship/FlyRank-AI-Internship


# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/rishh19/FlyRank-AI-Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

##1. My Lane

I chose the **Refresh / Content Opportunity Scoring** lane because I am interested in using machine learning to identify web pages that may need content updates. This project can help prioritize pages for review based on measurable signals such as impressions, CTR, content age, and traffic trends. I believe this is a practical problem where machine learning can support faster and better decision-making.

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Dataset loaded successfully.")
print("Total pages:", len(df))

Dataset loaded successfully.
Total pages: 30000


##2. The Question: Decision, Action, and Cost

**Research Question:**
Which web pages should be prioritized for content refresh based on observable performance signals such as impressions, CTR, content age, and traffic trends?

**Decision:**
This project aims to improve the decision of which pages should be reviewed first for content updates.

**Action:**
The SEO or content team can use the ranked recommendations to prioritize their limited time and focus on pages that are most likely to benefit from a refresh.

**Cost of a Wrong Recommendation:**
If the recommendations are incorrect, the team may spend valuable time reviewing pages that do not need updates, while genuinely declining pages may be missed. This can reduce the effectiveness of content maintenance and delay improvements to important pages.

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
priority_pages = df[df["trend_direction"] == "down"]

print("Pages that would be prioritized for review:", len(priority_pages))
print("Percentage of dataset:",
      round(len(priority_pages) / len(df) * 100, 2), "%")

Pages that would be prioritized for review: 16262
Percentage of dataset: 54.21 %


## 3.Quick Look at the Data

The starter dataset contains **30,000** web pages, of which **16,262** are currently marked as declining. The average content age is **256.17 days**. These numbers show that a large number of pages may require review, making it useful to build a machine learning system that helps prioritize which pages should be refreshed first.

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
total_pages = len(df)
declining_pages = (df["trend_direction"] == "down").sum()
avg_content_age = df["content_age_days"].mean()

print(f"Total pages: {total_pages}")
print(f"Declining pages: {declining_pages}")
print(f"Average content age: {avg_content_age:.2f} days")

Total pages: 30000
Declining pages: 16262
Average content age: 256.17 days


##4. What I Can and Cannot Claim

This project can identify patterns in the available data and provide decision-support for prioritizing content refresh. The results should be treated as observed and data-driven recommendations, not as proof of why pages decline. This project does not predict Google's algorithm or establish cause-and-effect relationships.

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("This project provides decision-support based on observed data.")
print("It does not predict Google's algorithm or prove causation.")

This project provides decision-support based on observed data.
It does not predict Google's algorithm or prove causation.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.